### DISCUSSION TITLE
Snowflake Support

### DISCUSSION THREAD

**Discussion by sfc-gh-kkeller** [SigNoz/signoz#1017] [Ideas]
I work for Snowflake and I would love to add Snowflake support to SigNoz as an OLAP engine. I am just trying to figure out where to start. Snowflake has a GOlang driver so I all I need to is write the backend support for this. 

Could you point me in the right direction of the general steps where I can start adding the support for it? 

Thanks!

**Comment by pranay01** [comment]
@kkellersnow Can you share more details on this. Currently SigNoz uses ClickHouse the main OLAP datastore. With this proposal, will SigNoz users be able to use Snowflake as the OLAP store instead of Clickhouse?

**Comment by sfc-gh-kkeller** [comment]
Yes, that would exactly be the goal. I made some progress in terms of where the code is that accesses clickhouse from the Query Service. Unfort. GoLang is not my strong point so I will struggle with this. While creating a bridge such as opentelemetry collector --> gRPC OLTP receiver --> ingestion to Snowflake is something I can do. If someone is interesed I can help from a Snowflake perspective though. I think it makes for a compelling use case.
